src/PersistentFilters.ts

```diff
@@ -160,68 +130,146 @@ export class PersistentFilters {
+
+    /**
+     * Empties the persistent search history array
+     * Sets the extension's searchHistory array to empty, and updates the array in settings.json
+     *
+     */
+    public async resetSearchHistory() {
+        this.mSearchHistory = [];
+        this.updateSearchHistory();
+    }
+
+    /**
+     * Empties the persistent sessions array
+     * Sets the extension's sessions array to empty, and updates the array in settings.json
+     *
+     */
     public async resetSessions() {
         this.mSessions = [];
         this.updateSessions();
     }
 
     /**
-     * Initializes the history and sessions sections by reading from a file
+     * Empties the persistent recently-opened files history array
+     * Sets the extension's fileHistory array to empty, and updates the array in settings.json
+     *
      */
-    private async initialize() {
-        let lines: string[];
-        let recallLines: string[];
-        if (vscode.workspace.getConfiguration(this.schema)) {
-            lines = vscode.workspace.getConfiguration(this.schema).get(PersistentFilters.history);
-            recallLines = vscode.workspace.getConfiguration(this.schema).get(PersistentFilters.recall);
-        }
-        if (recallLines) {
-            this.mRecall = recallLines;
-        }
-        if (lines) {
-            this.mHistory = lines;
-        } else {
-            this.resetHistory();
-        }
-        if (vscode.workspace.getConfiguration(this.schema)) {
-            lines = vscode.workspace.getConfiguration(this.schema).get(PersistentFilters.sessions);
-        }
-        if (lines) {
-            this.mSessions = lines;
-        } else {
-            this.resetSessions();
-        }
+    public async resetFileHistory() {
+        this.mFileHistory = [];
+        this.updateFileHistory();
     }
 
-    private async updateHistory() {
-        // settings are read-only, so make a clone
+    /*********************************************************************************************************************************************/
+    /* Update functions, for updating the settings.json file in VSCode
+    /*********************************************************************************************************************************************/
+
+    /**
+     * Updates the stored favorites array in settings.json
+     */
+    public async updateFavorites(favorites: string[]) {
+        // settings are read-only, so were cloned
         const settings: any = { ...vscode.workspace.getConfiguration(this.schema) };
         if (settings.persistence) {
-            settings.history = this.mHistory;
+            settings.favorites = favorites;
             await vscode.workspace.getConfiguration().update(this.schema, settings, vscode.ConfigurationTarget.Global);
         }
     }
 
-    private async updateRecall() {
+    /**
+     * Updates the search history array in settings.json
+     */
+    private async updateSearchHistory() {
```

Comment:
I think the block comment above + function name makes it clear and this comment is redundant.